lib/Dialect/TTNN/Utils/OptimizerOverrides.cpp

```diff
@@ -189,4 +189,335 @@ void InputLayoutOverrideParser::print(
+      options += "L1Interleaved ";
+      break;
+    }
+  }
+
+  // Create input layout overrides.
+  //  Example: insert-memreconfig=input0=0:1,input1=0,input2=0:1:2
+  if (inputLayoutOverrides.size() > 0) {
+    options += "insert-memreconfig=";
+    // Read operation name and operand indexes.
+    for (const auto &entry : inputLayoutOverrides) {
+      // Concatenate operation name.
+      options += std::string(entry.getKey()) + "=";
+      // Concatenate operand indexes.
+      const InputLayoutOverrideParams &params = entry.getValue();
+      // Concatenate operand indexes.
+      for (int64_t operandIdx : params.operandIdxes) {
+        options += std::to_string(operandIdx) + ":";
+      }
+      // Remove the last colon.
+      options.pop_back();
+      options += ",";
+    }
+    // Remove the last comma.
+    options.pop_back();
+    // Add a space for the next option.
+    options += " ";
+  }
+
+  // Create output layout overrides.
+  //  Example:
+  //  override-output-layout=op1=2x2:dram:interleaved:tile:fp32,op2=4x4:l1:block_sharded:row_major:fp16
+  //  Example:
+  //  override-output-layout=add_1_2=1x1:dram:interleaved:row_major:f32"
+  if (outputLayoutOverrides.size() > 0) {
+    options += "override-output-layout=";
+    // Read operation name and output layout parameters.
+    for (const auto &entry : outputLayoutOverrides) {
+      // Concatenate operation name.
+      options += std::string(entry.getKey()) + "=";
+      const OutputLayoutOverrideParams &params = entry.getValue();
+      // Concatenate grid size.
+      for (int64_t gridValue : params.grid) {
+        options += std::to_string(gridValue) + "x";
+      }
+      // Remove the last x.
+      options.pop_back();
+      // Concatenate buffer type
+      // enum class BufferType : uint32_t {
+      //   DRAM = 0,
+      //   L1 = 1,
+      //   SystemMemory = 2,
+      //   L1Small = 3,
+      //   Trace = 4,
+      // };
+      options += ":";
+      switch (params.bufferType) {
+      case mlir::tt::ttnn::BufferType::DRAM:
+        options += "dram";
+        break;
+      case mlir::tt::ttnn::BufferType::L1:
+        options += "l1";
+        break;
+      case mlir::tt::ttnn::BufferType::SystemMemory:
+        options += "system_memory";
+        break;
+      case mlir::tt::ttnn::BufferType::L1Small:
+        options += "l1_small";
+        break;
+      case mlir::tt::ttnn::BufferType::Trace:
+        options += "trace";
+        break;
+      }
+
+      // Concatenate tensor memory layout.
+      // enum class TensorMemoryLayout : uint32_t {
+      //   None = 0,
+      //   Interleaved = 1,
+      //   SingleBank = 2,
+      //   HeightSharded = 3,
+      //   WidthSharded = 4,
+      //   BlockSharded = 5,
+      // };
+      options += ":";
+      switch (params.tensorMemoryLayout) {
+      case mlir::tt::ttnn::TensorMemoryLayout::None:
+        options += "none";
+        break;
+      case mlir::tt::ttnn::TensorMemoryLayout::Interleaved:
```

Comment:
To these attributes have toString or print methods? Would scale much better, similar for types below.

Reply:
I'll check, if they don't exist I'll add. Also, this related to the answer on the previous question.

Reply:
Done.